server/src/main/java/org/apache/iotdb/db/engine/compaction/performer/impl/ReadPointCompactionPerformer.java

```diff
@@ -237,6 +236,63 @@ private void compactNonAlignedSeries(
     compactionWriter.endChunkGroup();
   }
 
+  private Map<String, MeasurementSchema> getMeasurementSchema(
+      String device, Set<String> measurements) throws IllegalPathException, IOException {
+    HashMap<String, MeasurementSchema> schemaMap = new HashMap<>();
+    List<TsFileResource> allResources = new LinkedList<>(seqFiles);
+    allResources.addAll(unseqFiles);
+    for (String measurement : measurements) {
+      for (TsFileResource tsFileResource : allResources) {
```

Comment:
better to scan from the last, a previous resource may have a deleted chunk with different datatype